code/scripts/components/snippetGenerator/index.js

```diff
@@ -16,31 +16,33 @@ class SnippetGenerator extends React.Component {
 
     this.handleEmailChange = this.handleEmailChange.bind(this)
     this.handleButtonClick = this.handleButtonClick.bind(this)
+    this.handleEmailBlur = this.handleEmailBlur.bind(this)
   }
 
   handleEmailChange(e) {
     const email = e.currentTarget.value
     this.setState({ email })
+  }
 
-    if (window.mixpanel) {
-      mixpanel.track("Entered Email in Snippet Generator");
-    }
+  handleEmailBlur(e) {
+    const email = e.currentTarget.value
+    mixpanel.identify(email);
+    mixpanel.people.set({'$email': email});
+    mixpanel.track('Entered email in snippet generator')
   }
 
   handleButtonClick(e) {
     const { email } = this.state
 
+    mixpanel.track('Clicked generate snippet button on snippet generator')
```

Comment:
Everywhere in this snippet generator component (and other components) that we're calling `mixpanel.track`, we should first check to see that `mixpanel` exists before calling it. A not insignificant number of people browse with tracking blockers enabled these days, which would cause this script to error at this point and prevent the user from being able to generate a snippet.